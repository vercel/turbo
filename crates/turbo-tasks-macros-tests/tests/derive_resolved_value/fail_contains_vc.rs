#![allow(dead_code)]

use turbo_tasks::{ResolvedValue, ResolvedVc, Vc};

#[derive(ResolvedValue)]
struct ContainsOnlyVc {
    a: Vc<i32>,
}

#[derive(ResolvedValue)]
struct ContainsResolvedVcAndVc {
    a: ResolvedVc<i32>,
    b: Vc<i32>,
}

#[derive(ResolvedValue)]
struct ContainsVcInsideGeneric {
    a: Option<Box<[Vc<i32>; 4]>>,
}

fn main() {}
