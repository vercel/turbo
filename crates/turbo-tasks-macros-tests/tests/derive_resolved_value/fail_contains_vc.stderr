error[E0277]: the trait bound `Vc<i32>: ResolvedValue` is not satisfied
  --> tests/derive_resolved_value/fail_contains_vc.rs:13:8
   |
13 |     b: Vc<i32>,
   |        ^^^^^^^ the trait `ResolvedValue` is not implemented for `Vc<i32>`
   |
   = help: the following other types implement trait `ResolvedValue`:
             &str
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
             (T,)
           and $N others
note: required by a bound in `_::{closure#0}::DeriveResolvedValueAssertion::assert_impl_resolved_value`
  --> tests/derive_resolved_value/fail_contains_vc.rs:10:10
   |
10 | #[derive(ResolvedValue)]
   |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
   = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Vc<i32>: ResolvedValue` is not satisfied
 --> tests/derive_resolved_value/fail_contains_vc.rs:7:8
  |
7 |     a: Vc<i32>,
  |        ^^^^^^^ the trait `ResolvedValue` is not implemented for `Vc<i32>`
  |
  = help: the following other types implement trait `ResolvedValue`:
            &str
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
            (T,)
          and $N others
note: required by a bound in `_::{closure#0}::DeriveResolvedValueAssertion::assert_impl_resolved_value`
 --> tests/derive_resolved_value/fail_contains_vc.rs:5:10
  |
5 | #[derive(ResolvedValue)]
  |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
  = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Vc<i32>: ResolvedValue` is not satisfied
  --> tests/derive_resolved_value/fail_contains_vc.rs:18:8
   |
18 |     a: Option<Box<[Vc<i32>; 4]>>,
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ResolvedValue` is not implemented for `Vc<i32>`, which is required by `Option<Box<[Vc<i32>; 4]>>: ResolvedValue`
   |
   = help: the following other types implement trait `ResolvedValue`:
             &str
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
             (T,)
           and $N others
   = note: required for `[Vc<i32>; 4]` to implement `ResolvedValue`
   = note: 2 redundant requirements hidden
   = note: required for `Option<Box<[Vc<i32>; 4]>>` to implement `ResolvedValue`
note: required by a bound in `_::{closure#0}::DeriveResolvedValueAssertion::assert_impl_resolved_value`
  --> tests/derive_resolved_value/fail_contains_vc.rs:16:10
   |
16 | #[derive(ResolvedValue)]
   |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
   = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)
